src/linear-client.ts

```diff
@@ -0,0 +1,351 @@
+import assert from 'assert';
+import { AxiosRequestConfig } from 'axios';
+import { GenericAPIResponse, RestClientInverseOptions } from './util/requestUtils';
+import RequestWrapper from './util/requestWrapper';
+import { SharedEndpoints } from './shared-endpoints';
+
+class _LinearClient {
+  private requestWrapper: RequestWrapper;
+
+  /**
+   * @public Creates an instance of the inverse REST API client.
+   *
+   * @param {string} key - your API key
+   * @param {string} secret - your API secret
+   * @param {boolean} [livenet=false]
+   * @param {RestClientInverseOptions} [restInverseOptions={}] options to configure REST API connectivity
+   * @param {AxiosRequestConfig} [requestOptions={}] HTTP networking options for axios
+   */
+  constructor(
+    key?: string | undefined,
+    secret?: string | undefined,
+    livenet?: boolean,
+    restInverseOptions:RestClientInverseOptions = {}, // TODO: Rename this type to be more general.
+    requestOptions: AxiosRequestConfig = {}
+  ) {
+    this.requestWrapper = new RequestWrapper(key, secret, livenet, restInverseOptions, requestOptions);
+  }
+
+  // __ Market Data Endpoints
+  // https://bybit-exchange.github.io/docs/linear/#t-marketdata
+
+  /**
+   * @public Getch candle data for a symbol.
+   */
+  getKline(params: any): GenericAPIResponse {
+    assert(params, 'No params passed');
+    assert(params.symbol, 'Parameter symbol is required');
+    assert(params.interval, 'Parameter interval is required');
+    assert(params.from, 'Parameter from is required');
```

Comment:
These assert calls can be dropped, see how it is now in the rest-client in master